client/src/components/header/SmallDisplayStepModalHeader.vue

```diff
@@ -4,59 +4,45 @@
   <ion-button
     v-if="!hideCloseButton"
     slot="icon-only"
-    class="closeBtn"
-    :class="{
-      'closeBtn-stepper': step,
-    }"
+    class="closeBtn closeBtn-stepper"
     @click="$emit('closeClicked')"
   >
     <ion-icon
       :icon="close"
       class="closeBtn__icon"
     />
   </ion-button>
-  <ion-header
-    class="modal-header"
-    :class="{
-      'modal-header-stepper': step,
-    }"
-  >
-    <div
-      class="modal-header-stepper-label"
-      v-if="step"
-    >
+  <ion-header class="modal-header">
+    <div class="modal-header-stepper">
       <ion-icon
-        :icon="step.icon"
-        class="modal-header-stepper-label__icon"
+        :icon="icon"
+        class="modal-header-stepper__icon"
       />
-      <ion-text class="modal-header-stepper-label__text button-medium">
-        {{ $msTranslate(step.title) }}
+      <ion-text class="modal-header-stepper__text button-medium">
+        {{ $msTranslate(title) }}
       </ion-text>
     </div>
-    <div class="modal-header-title">
-      <ion-text
-        class="modal-header-title__step body"
-        v-if="step?.current !== undefined"
-      >
+    <div class="modal-header-content">
+      <ion-text class="modal-header__step body">
         {{
           $msTranslate({
             key: 'HeaderPage.modalHeader.step',
             data: {
-              current: step.current,
-              total: step.total,
+              current: currentStep,
+              total: steps.length,
             },
           })
         }}
       </ion-text>
-      <ion-text class="modal-header-title__text title-h2">
-        {{ $msTranslate(title) }}
+      <ion-text class="modal-header__title title-h2">
+        {{ $msTranslate(steps[currentStep - 1].title) }}
```

Comment:
The parent component should pass the right step index, you should not have to guess.